src/app/pages/quizzes-catalog/quizzes-catalog.component.ts

```diff
@@ -7,45 +7,30 @@ import {
   cardItemStyles,
 } from '../../ui-kit/constants/card-item';
 import { CategoryModel } from '../../services/model/category.model';
-import { catchError, map, Subscription } from 'rxjs';
-import { CategoryService } from '../../services/catalog.service';
+import { Observable } from 'rxjs';
+import { Store } from '@ngrx/store';
+import * as CategoryActions from '../../../store/category/category.actions';
+import { AppState } from '../../../store/app.store';
 
 @Component({
   standalone: true,
   selector: 'app-quizzes-catalog',
   templateUrl: './quizzes-catalog.component.html',
   imports: [ UIKitModule, CommonModule ],
 })
-export class QuizzesCatalogComponent implements OnInit, OnDestroy {
-  quizzesCatalog: CategoryModel[] = [];
-  errorMessage: string = '';
-  showSpinner: boolean = false;
-
-  subscription!: Subscription;
-
-  constructor(private categoriesService: CategoryService) {}
-
-  ngOnInit(): void {
-    this.showSpinner = true;
-    this.subscription = this.categoriesService.get().pipe(
-      map(data => this.quizzesCatalog = data),
-      catchError(error => this.errorMessage = error)
-    ).subscribe();
-
-    setTimeout(() => {
-      this.showSpinner = false;
-    }, 2000);
+export class QuizzesCatalogComponent {
+  categories$: Observable<CategoryModel[]>;
+  isLoading$: Observable<boolean>;
+  
+  constructor(private store: Store<AppState>) {
+    this.categories$ = this.store.select(state => state.categories.categories);
+    this.isLoading$ = this.store.select(state => state.categories.loading);
+    this.store.dispatch(CategoryActions.loadCategories());
```

Comment:
can these be moved to the OnInit?
And I'd rather create additional service who is responsible for communication with store. This service can have methods that can be invoked here. Is it possible?